bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -675,5 +723,226 @@ private void validateSQL(String sql) throws TskCoreException {
 		 * TODO (JIRA-5950): Need SQL injection defense in CaseDbAccessManager 
 		 */
 	}
+	
+	/**
+	 * A wrapper around a PreparedStatement to execute queries against the
+	 * database.
+	 */
+	@Beta
+	public static class CasePreparedStatement implements AutoCloseable {
+
+		private final CaseDbConnection connection;
+		private final PreparedStatement preparedStatement;
+
+		/**
+		 * Main constructor.
+		 *
+		 * @param connection        The db connection.
+		 * @param preparedStatement The delegate prepared statement.
+		 */
+		CasePreparedStatement(CaseDbConnection connection, PreparedStatement preparedStatement) {
```

Comment:
It looks to me like this constructor should be private. I believe only the enclosing class needs to create instances of CasePreparedStatement objects.